felix/bpf/ut/nat_test.go

```diff
@@ -3712,3 +3712,53 @@ func TestNATNodePortV6(t *testing.T) {
 		}, withHostNetworked(), withIPv6())
 	}
 }
+
+// TestNATOutExcludeHosts tests the NATOutgoingExclusions option in FelixConfig. When
+// it's set to 'IPPoolsAndHostIPs', the CALI_GLOBALS_NATOUTGOING_EXCLUDE_HOSTS global
+// flag will be set and packets to any hosts should not be SNAT'ed.
+func TestNATOutExcludeHosts(t *testing.T) {
+	RegisterTestingT(t)
+	_, ipv4, l4, _, pktBytes, err := testPacketUDPDefault()
+	Expect(err).NotTo(HaveOccurred())
+	udp := l4.(*layers.UDP)
+
+	hostIP = node1ip
+	// Insert a reverse route for the source workload (flagged in IP pool and NAT-outgoing).
+	rtKey := routes.NewKey(srcV4CIDR).AsBytes()
+	rtVal := routes.NewValueWithIfIndex(routes.FlagsLocalWorkload|routes.FlagInIPAMPool|routes.FlagNATOutgoing, 1).AsBytes()
+
+	err = rtMap.Update(rtKey, rtVal)
+	Expect(err).NotTo(HaveOccurred())
+	// Insert a route for the dest ip (flagged as remote host).
+	rtKey = routes.NewKey(dstV4CIDR).AsBytes()
+	rtVal = routes.NewValue(routes.FlagsRemoteHost).AsBytes()
+
+	err = rtMap.Update(rtKey, rtVal)
+	Expect(err).NotTo(HaveOccurred())
+	defer resetRTMap(rtMap)
+	_ = conntrack.NewKey(uint8(ipv4.Protocol),
+		ipv4.SrcIP, uint16(udp.SrcPort), ipv4.DstIP, uint16(udp.DstPort))
```

Comment:
you never use this - you could use that key to make a lookup in ct map to see whether `v4.FlagNATOut` is set on the value. But I think testing the mark is good enough.